os/kernel/task/task_activate.c

```diff
@@ -104,7 +105,14 @@ void thread_termination_handler(int signo, siginfo_t *data)
 
 #ifdef CONFIG_SIGKILL_HANDLER
 	if (rtcb->sigkillusrhandler != NULL) {
-		rtcb->sigkillusrhandler(signo, data, NULL);
+		if ((rtcb->flags & TCB_FLAG_TTYPE_MASK) >> TCB_FLAG_TTYPE_SHIFT == TCB_FLAG_TTYPE_KERNEL) {
+			rtcb->sigkillusrhandler(signo, data, NULL);
+		} else {
+			siginfo_t info;
+			memcpy(&info, data, sizeof(siginfo_t));
+
+			up_signal_dispatch(rtcb->sigkillusrhandler, signo, &info, NULL);
```

Comment:
Please check the below conditions.
`#if defined(CONFIG_BUILD_PROTECTED) || defined(CONFIG_BUILD_KERNEL)`

Reply:
Thanks, I added.